spark-connector/spark-common/build.gradle.kts

```diff
@@ -171,3 +174,18 @@ configurations {
 artifacts {
   add("testArtifacts", testJar)
 }
+
+tasks.withType<ShadowJar>(ShadowJar::class.java) {
```

Comment:
we could shade the jar in spark-runtime module, like `spark-connector/v3.5/spark-runtime/build.gradle.kts`

Reply:
Do we need to add this to every spark-runtime version package? Should we extract the common packages or handle them directly in the common module? This way, we can avoid redundant processing.

Reply:
I would be inclined to use @FANNG1 's way to shade jars in spark-runtime, so we don't have to do the shade in this `common` component, and change all the `implementation` to `shadow`.

Reply:
ok

Reply:
fixed